pkg/collector/corechecks/cluster/kubernetesapiserver/kubernetes_eventbundle.go

```diff
@@ -133,18 +133,27 @@ func (b *kubernetesEventBundle) formatEvents(clusterName string, providerIDCache
 		Tags:           tags,
 		AggregationKey: fmt.Sprintf("kubernetes_apiserver:%s", b.objUID),
 		AlertType:      b.alertType,
-		Text: fmt.Sprintf(
-			"%%%%%% \n%s \n _Events emitted by the %s seen at %s since %s_ \n\n %%%%%%",
-			formatStringIntMap(b.countByAction),
-			b.component,
-			time.Unix(int64(b.lastTimestamp), 0),
-			time.Unix(int64(b.timeStamp), 0),
-		),
+		Text:           b.formatEventText(),
 	}
 
 	return output, nil
 }
 
+func (b *kubernetesEventBundle) formatEventText() string {
+	eventText := fmt.Sprintf(
+		"%%%%%% \n%s \n _Events emitted by the %s seen at %s since %s_ \n\n %%%%%%",
+		formatStringIntMap(b.countByAction),
+		b.component,
+		time.Unix(int64(b.lastTimestamp), 0),
+		time.Unix(int64(b.timeStamp), 0),
+	)
+
+	// Escape the ~ character to not strike out the text
+	eventText = strings.ReplaceAll(eventText, "~", "\\~")
```

Comment:
You'll need to add a unit test for that. Also I wonder about the performance impact, it would be nice to have a small benchmark on typical event size.